WebCryptoAPI/sign_verify/ecdsa.js

```diff
@@ -0,0 +1,400 @@
+                });
+            }, vector.name + " using publicKey to sign");
+        }, function(err) {
+            promise_test(function(test) {
+                assert_unreached("importVectorKeys failed for " + vector.name + ". Message: ''" + err.message + "''");
+            }, "importVectorKeys step: " + vector.name + " using publicKey to sign");
+        });
+
+        all_promises.push(promise);
+    });
+
+    // Check for failures due to no "verify" usage.
+    testVectors.forEach(function(originalVector) {
+        var vector = Object.assign({}, originalVector);
+
+        var promise = importVectorKeys(vector, [], ["sign"])
+        .then(function(vectors) {
+            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
+            promise_test(function(test) {
+                return subtle.verify(algorithm, vector.publicKey, vector.signature, vector.plaintext)
+                .then(function(plaintext) {
+                    assert_unreached("Should have thrown error for no verify usage in " + vector.name + ": " + err.message + "'");
+                }, function(err) {
+                    assert_equals(err.name, "InvalidAccessError", "Should throw InvalidAccessError instead of '" + err.message + "'");
+                });
+            }, vector.name + " no verify usage");
+        }, function(err) {
+            promise_test(function(test) {
+                assert_unreached("importVectorKeys failed for " + vector.name + ". Message: ''" + err.message + "''");
+            }, "importVectorKeys step: " + vector.name + " no verify usage");
+        });
+
+        all_promises.push(promise);
+    });
+
+    // Check for successful signing and verification.
+    testVectors.forEach(function(vector) {
+        var promise = importVectorKeys(vector, ["verify"], ["sign"])
+        .then(function(vectors) {
+            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
+            promise_test(function(test) {
+                return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
+                .then(function(signature) {
+                    // Can we get verify the signature?
+                    return subtle.verify(algorithm, vector.publicKey, signature, vector.plaintext)
+                    .then(function(is_verified) {
+                        assert_true(is_verified, "Round trip verification works");
+                        return signature;
+                    }, function(err) {
+                        assert_unreached("verify error for test " + vector.name + ": " + err.message + "'");
+                    });
+                })
+                .then(function(priorSignature) {
+                    // Will a second signing give us different signature, as it should?
+                    return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
```

Comment:
I agree. I'll get right on it.

Charlie

On Fri, Jun 17, 2016 at 6:05 PM, Jim Schaad <notifications@github.com>
wrote:

> In WebCryptoAPI/sign_verify/ecdsa.js
> <https://github.com/w3c/web-platform-tests/pull/3183#discussion_r67581916>
> :
>
> > +            var algorithm = {name: vector.algorithmName, hash: vector.hashName};
> > +            promise_test(function(test) {
> > +                return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
> > +                .then(function(signature) {
> > +                    // Can we get verify the signature?
> > +                    return subtle.verify(algorithm, vector.publicKey, signature, vector.plaintext)
> > +                    .then(function(is_verified) {
> > +                        assert_true(is_verified, "Round trip verification works");
> > +                        return signature;
> > +                    }, function(err) {
> > +                        assert_unreached("verify error for test " + vector.name + ": " + err.message + "'");
> > +                    });
> > +                })
> > +                .then(function(priorSignature) {
> > +                    // Will a second signing give us different signature, as it should?
> > +                    return subtle.sign(algorithm, vector.privateKey, vector.plaintext)
>
> I would be better with removing it. The text probably says choose k at
> random or psuedorandom. I would say that making it deterministic might fit
> into the psuedorandom case, but in any event many people are starting to
> implement it because of the better security practices.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/3183/files/7f598aa6098a188859bea290f147c48b5712e999#r67581916>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAE_N8agykEZQ0aSW40SzZHecAgSE9Eaks5qMxoqgaJpZM4I2ltT>
> .
>